src/field/arena.ts

```diff
@@ -274,6 +274,12 @@ export class Arena {
 
     const oldWeatherType = this.weather?.weatherType || WeatherType.NONE;
 
+    if (this.weather?.isImmutable() && (![ WeatherType.HARSH_SUN, WeatherType.HEAVY_RAIN, WeatherType.STRONG_WINDS, WeatherType.NONE ].includes(weather))) {
```

Comment:
```suggestion
    if (this.weather?.isImmutable() && ![ WeatherType.HARSH_SUN, WeatherType.HEAVY_RAIN, WeatherType.STRONG_WINDS, WeatherType.NONE ].includes(weather)) {
```
Oops, forgot to remove the `()` in the last suggestion.